hooks/nrpe_helpers.py

```diff
@@ -25,6 +25,15 @@
 DB_KEY_KNOWN_REBOOT_TIME = "known_reboot_time"
 db = unitdata.kv()
 
+# These are the valid options to keep compatibility on changes between bionic and focal
+VALID_CHARM_CIS_PROFILES = [
+    "",
+    "level1_server",
+    "level2_server",
+    "level1_workstation",
+    "level2_workstation",
+]
```

Comment:
You'll need to add logic to `cis_cmd_params()`, to convert from these profiles to focal compatible profile names - eg.:

```
# pseudocode
if system >= focal:
  profile = old_to_new_profile_mapping[profile]
```

Reply:
I was doing that in my first approach,  but IMO this adds an extra complexity without necessity.

The only part that really needs to map those profiles are the one that calls the CIS binary. The CLI for cronjob and the  check itself can keep the options fixed list without the `cis_` prefix. This map is happening at `_get_cis_hardening_profile`.

If we want to allow both options be valid charm config, e.g: `level1_server` and `cis_level1_server` then we'll need to always find the right options for the nrpe check and for the cron job. I prefer to keep it simple and at a single place. This will make easier for maintenance and to understand.

Reply:
I think we should translate as late as possible in the process, i.e. within nrpe only speak "legacy" profile names, and add the `cis_` prefix at check render time, if needed.
If at some point we need to move past the CIS profiles, I think we should create a separate option (e.g. `usg_profiles`) and finally abandon this codepath (and bionic).

Reply:
Sure, I don't mind where we add the translation mapping logic, but it does need to be added; I don't see any mapping happening in this PR yet.

Reply:
> This map is happening at _get_cis_hardening_profile.

I don't see a mapping - it validates the profile based on which ubuntu release is running, but where does it map from the the charm config option to the correct one for the ubuntu release?

Reply:
> I don't see a mapping

Glad that you catch that, thanks. It was indeed not mapping in the end and I took the structure from the unit tests that was accessing profiles by position like `focal_profiles[0]` and that mislead me. I've changed the unit tests now to be more declarative and show that e.g. `level2_workstation` will turn into `cis_level2_workstation` on `_get_cis_hardening_profile`